### PR TITLE
Remove more dead code

### PR DESCRIPTION
**Pull request by ipince**
## Summary
TSIA

## How was it tested?
CICD

### CODE REVIEW

File: internal/impl/shell.go [hunk 1]
```diff
@@ -289,17 +285,10 @@ func (s *DevboxShell) Run() error {
 	// This could be from one of the generated shellrc commands, but more likely is from
 	// a user's command or script. So, we want to return nil for this.
 	if exitErr := (&exec.ExitError{}); errors.As(err, &exitErr) {
-
-		// The exception to the previous comment is if we are executing a shell script
-		// via `devbox run` or the deprecated `devbox shell -- <command>`. In this case,
-		// we do want to return the exit code of the script that was run.
```
Comment (savil) [inline internal/impl/shell.go:295]: do we still have this behavior (elsewhere) of returning the exit code from a user's script?

File: internal/impl/shell.go [hunk 2]
```diff
@@ -354,11 +343,6 @@ func (s *DevboxShell) writeDevboxShellrc() (path string, err error) {
 		}
 	}()
 
-	pathPrepend := s.profileDir + "/bin"
-	if s.pkgConfigDir != "" {
-		pathPrepend = s.pkgConfigDir + ":" + pathPrepend
-	}
```
Comment (savil) [inline internal/impl/shell.go:360]: I understand that the profileDir should no longer be added to PATH
but why remove the plugins' pkgConfigDir?

**Review by savil — COMMENTED** [review]
I may be missing something, but these look like logic changes rather than dead code?

File: internal/impl/shell.go (hunk 1 above)
Comment (ipince) [reply internal/impl/shell.go:295]: Executing scripts (i.e. `devbox run`) doesn't happen through this code path. So, basically, `s.ScriptCommand` is _always_ `""`

Comment (ipince) [reply internal/impl/shell.go:295]: And yes, we do propagate the exit code from a script to the user (see `nix/run.go`)

File: internal/impl/shell.go (hunk 2 above)
Comment (ipince) [reply internal/impl/shell.go:360]: `pathPrepend` is only used when `ExportEnv` is empty. But `ExportEnv` is never empty ever since unified env is on.

**Review by savil — APPROVED** [review]
got it, thanks for the explanations!

**Pull request merged**